### ISSUE TITLE
[Bug Report] Gym environment compliance API makes false seeding assumption

### ISSUE THREAD

**Issue by JesseFarebro** [openai/gym#2892]
If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
When you run the Gym compliance check it assumes that `.reset()` will alter the state of `np_random`. If it doesn't it fails the check here: https://github.com/openai/gym/blob/053ee809324861dc1bec19feee97b47ba1557a94/gym/utils/env_checker.py#L79-L86

**Code example**
Please try to provide a minimal example to reproduce the bug. Error messages and stack traces are also helpful.

```py

class TestEnv(gym.Env):

    def reset(self, *, seed: Optional[int] = None, return_info: Optional[bool] = False, options: Optional[dict[str, Any]] = None):
        pass
```

Not a complete code example but the environment above should pass compliance checks.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/openai/gym/issues) in the repo (**required**)

**Comment by pseudo-rnd-thoughts** [comment]
That shouldn't pass the environment checker as the seed parameter is ignored.
We assume a minimal environment should look like 
```python
class MinEnv(gym.Env):

    def __init__(self):
        self.observation_space = spaces.Box(0, 1, (1,))
        self.action_space = spaces.Box(0, 1, (1,))

    def reset(self, *, seed: Optional[int] = None, return_info: bool = False, options: Optional[dict] = None) -> Union[ObsType, Tuple[ObsType, dict]]:
        super().reset(seed=seed)

        if return_info:
            return self.observation_space.sample(), {}
        else:
            return self.observation_space.sample()
```

Why do you think the seed parameter should be ignored?

**Comment by RedTachyon** [comment]
Unless I'm confusing something, this is something we already discussed @pseudo-rnd-thoughts, right? It's currently being fixed in #2873.

@JesseFarebro I'm not sure to what extent you wanted to "not-complete-code" your example, but the key part it's missing here is the `super()` call in `reset`. You're also right that the current env checker will erroneously raise an exception for a minimal implementation that has it. 

In the [new version of the checker](https://github.com/openai/gym/pull/2873/files#diff-5f10831b2f9a99f913f1fcfe4ba9f701ddbd15afc57bbafeb553793070da448bR94), we're explicitly reseeding the environment. If it calls `super()` correctly, then `np_random` will get updated (even if that's somewhat redundant in this specific case) and the state will indeed change.

**Comment by JesseFarebro** [comment]
Here, I'll provide a complete example this time. Sorry, I didn't have time earlier so I just wrote that to give you an idea.

```py
import functools
import gym
from gym import spaces
from gym.utils import env_checker
import numpy as np


class TestEnv(gym.Env):
    def __init__(self):
        pass

    def reset(self, *, seed: int | None = None, return_info: bool = False, options: dict | None = None):
        super().reset(seed=seed)

        if return_info:
            return np.zeros(4), {}
        else:
            return np.zeros(4)

    def step(self, action: int):
        return np.full((4,), action), 1, True

    @functools.cached_property
    def action_space(self):
        return spaces.Discrete(n=4)

    @functools.cached_property
    def observation_space(self):
        return spaces.Box(low=0, high=3, shape=(4,), dtype=np.float64)


env_checker.check_env(TestEnv())
```

The problem right now is that if you don't use `np_random` it doesn't matter if you call super correctly or not that assert will fail because it assumes that you'll generate a new RNG state when calling `reset(seed=None)`. This isn't true for the ALE and I'm sure isn't true for a whole bunch of other environments (e.g., the one I have above).

**Comment by pseudo-rnd-thoughts** [comment]
@JesseFarebro You are correct, this was a bug introduced in `v24.1` that has been patched in #2873 
However, we don't believe that we will be releasing a `v25`. We will be adding a note to the release notes for `v24.1` but due to the number of issues with `v24.x` I would skip it

**Comment by jkterry1** [comment]
Closing in favor of #2903
